### PR TITLE
Fix failing tests in ROCm

### PR DESCRIPTION
**Pull request by kmaehashi**
```
13:53:34 FAILED cupyx_tests/scipy_tests/special_tests/test_beta.py::TestBeta::test_beta_specific_vals
13:53:34 FAILED cupyx_tests/scipy_tests/special_tests/test_beta.py::TestBeta::test_betaln_specific_vals
```

Tolerance needs to be adjusted.

```
13:53:34 FAILED cupyx_tests/scipy_tests/special_tests/test_gammainc.py::TestGammainc::test_roundtrip
```
This issue seems to be resolved in ROCm 5.0.

### CODE REVIEW

**Comment by kmaehashi** [comment]
pfnCI, test this please.

File: tests/cupyx_tests/scipy_tests/special_tests/test_gammainc.py [hunk 1]
```diff
@@ -84,6 +84,10 @@ def gammainc_line(self, x):
     #     dataset = cp.vstack((a, x, self.gammainc_line(x))).T
     #     FuncData(sc.gammainc, dataset, (0, 1), 2, rtol=1e-11).check()
 
+    @pytest.mark.skipif(
+        cp.cuda.runtime.is_hip and
+        cp.cuda.runtime.runtimeGetVersion() < 5_00_00000,
+        reason='ROCm/HIP fails in ROCm 4.x')
```
Comment (emcastillo) [inline tests/cupyx_tests/scipy_tests/special_tests/test_gammainc.py:90]: I just tested this with rocm 5.0.1 and almost all the tests fails :(.
Maybe we should skip the whole gammainc tests for hip ...

Comment (kmaehashi) [reply tests/cupyx_tests/scipy_tests/special_tests/test_gammainc.py:90]: Oh, when I tested this passed with ROCm 5.0.1. (note that you will need #6466 for ROCm 5.0)

Comment (emcastillo) [reply tests/cupyx_tests/scipy_tests/special_tests/test_gammainc.py:90]: I just checked with the atomic header changes. I will re-run with #6466 fully applied! thanks.

However, the test that causes 4.3 to crash is a different one :)

Comment (emcastillo) [reply tests/cupyx_tests/scipy_tests/special_tests/test_gammainc.py:90]: master with that branch applied:
```
tests/cupyx_tests/scipy_tests/special_tests/test_gammainc.py .......F.FFF.F.F.......FFFFF
```

Comment (emcastillo) [reply tests/cupyx_tests/scipy_tests/special_tests/test_gammainc.py:90]: worked after clearing cache and compile from 0 everything

Comment (emcastillo) [reply tests/cupyx_tests/scipy_tests/special_tests/test_gammainc.py:90]: List of tests that are broken in rocm 4.3
```
TestGammainc::test_domain[nan-1]
TestGammainc::test_infinite_limits
TestGammainc::test_limit_check
TestGammainc::test_roundtrip
TestGammaincc::test_infinite_limits
TestGammaincc::test_roundtrip
```
These work in rocm 5.0.1 so should be skipped too :)

**Comment by emcastillo** [comment]
pfnCI, run Jenkins please.

**Comment by emcastillo** [comment]
pfnCI, run Jenkins please.

**Review by kmaehashi — COMMENTED** [review]
LGTM!

**Review by emcastillo — APPROVED** [review]

**Comment by kmaehashi** [comment]
No backport needed as v10 does not have these features.

**Pull request merged**